### PR TITLE
Order

### PR DESCRIPTION
**Pull request by babchukbohdan**

### CODE REVIEW

File: client/src/actions/cityInfoActions.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import { CityInfoValues, InfoActionTypes } from '../types/cityInfoTypes';
+import { CityInfoI } from '../pages/Order/mapService';
+
+export const getCityInfoCreator = (cityName: string) => {
+  console.log('action get info', cityName);
```
Comment (Pr1sm0) [inline client/src/actions/cityInfoActions.ts]: Delete

File: client/src/components/ExtraServiceItem.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        title={showTitle && name}
+        classNames={['extra-service__label']}
+        htmlFor={name}
+      >
+        <Form.Check
+          hidden
+          id={name}
+          aria-label={name}
+          type="checkbox"
+          // label={name}
```
Comment (Pr1sm0) [inline client/src/components/ExtraServiceItem.tsx]: Delete or uncomment if necessary

File: client/src/components/ExtraServicesSelect.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import React, { FC } from 'react';
+import { Accordion, Card, Form } from 'react-bootstrap';
+import { ExtraServicesI } from '../pages/Order/mapService';
+import { ExtraServiceItem } from './ExtraServiceItem';
+// Icons
```
Comment (Pr1sm0) [inline client/src/components/ExtraServicesSelect.tsx]: Same as above

File: client/src/components/ExtraServicesSelect.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+export const ExtraServicesSelect: FC<ExtraServicesSelectI> = ({
+  title,
+  avaliableInCityExtraServices,
+  activeExtraServices,
+  onExtraServicesChanged,
+}) => {
+  return avaliableInCityExtraServices ? (
+    <Accordion>
+      <Accordion.Toggle as={Form.Label} variant="link" eventKey="0">
+        {title || 'Extra services:'}
```
Comment (Pr1sm0) [inline client/src/components/ExtraServicesSelect.tsx]: Extra services to constants

File: client/src/http/userApi.ts [hunk 5]
```diff
@@ -2,26 +2,23 @@ import { $host, $authHost } from './index';
...
-    password,
-    role: 'USER',
-  });
+export const registration =
+  (name: string, phone: string, password: string) => async () => {
+    const { data } = await $host.post('user/registration', {
+      name,
+      phone,
+      password,
+      role: 'USER',
```
Comment (Pr1sm0) [inline client/src/http/userApi.ts]: To constant

File: client/src/pages/Order/Map.tsx [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+              <Marker key={`${marker.lat}${marker.lng}`} position={marker} />
+            ))} */}
+
+        {/* {currentLocation && (
+            <InfoWindow position={currentLocation}>
+              <div>
+                <p>Your location</p>
+              </div>
+            </InfoWindow>
+          )} */}
```
Comment (Pr1sm0) [inline client/src/pages/Order/Map.tsx]: Delete or uncomment if necessary

File: client/src/pages/Order/Map.tsx [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+  // TODO save renderer here
+
+  // const directionsServiceLoaded = useCallback((dirService: any) => {
+  //   // console.log('dirService', dirService);
+  // }, []);
+
+  const renderMap = () => {
+    // wrapping to a function is useful in case you want to access `window.google`
+    // to eg. setup options or create latLng object, it won't be available otherwise
+    // feel free to render directly if you don't need that
```
Comment (Pr1sm0) [inline client/src/pages/Order/Map.tsx]: Delete or uncomment if necessary

File: client/src/pages/Order/MapContainer/MapContainer.tsx [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+
+export const MapContainer = () => {
+  const { directions, directionsResult, currentLocation } = useTypedSelector(
+    (state) => state.map,
+  );
+  const { changeMapValue } = useMapActions();
+  const { changeOrderValue } = useOrderActions();
+
+  const [renderer, setrenderer] = useState<google.maps.DirectionsRenderer>();
+  // console.log('renderer', renderer);
```
Comment (Pr1sm0) [inline client/src/pages/Order/MapContainer/MapContainer.tsx]: Delete

File: client/src/pages/Order/MapContainer/MapContainer.tsx [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+        changeMapValue('directionsResult', result);
+      }
+    },
+    [],
+  );
+
+  const onMapLoaded = React.useCallback(function onLoad(mapInstance) {
+    // do something with map Instance
+
+    // changeMapValue('map', mapInstance);
```
Comment (Pr1sm0) [inline client/src/pages/Order/MapContainer/MapContainer.tsx]: Delete

File: client/src/pages/Order/MapContainer/MapContainer.tsx [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+      if (renderer) {
+        const directions =
+          renderer.getDirections() as google.maps.DirectionsResult;
+        const origin = directions.routes[0].legs[0].start_address;
+        const destination = directions.routes[0].legs[0].end_address;
+        const directionRoutes = directions.routes[0].legs[0];
+        const distance: google.maps.Distance =
+          directionRoutes.distance as google.maps.Distance;
+
+        console.log(distance, 'distance');
```
Comment (Pr1sm0) [inline client/src/pages/Order/MapContainer/MapContainer.tsx]: Delete

File: client/src/pages/Order/MapContainer/MapContainer.tsx [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+        changeOrderValue('distance', distance);
+        changeOrderValue('from', origin);
+        changeOrderValue('to', destination);
+      }
+    },
+    [renderer, directionsResult],
+  );
+
+  const onDirectionsRendererLoaded = useCallback((dirRenderer: any) => {
+    console.log('on renderer loaded');
```
Comment (Pr1sm0) [inline client/src/pages/Order/MapContainer/MapContainer.tsx]: Delete

File: client/src/pages/Order/Order.tsx [hunk 12]
```diff
@@ -1,13 +1,102 @@
...
+  };
+
+  const onFromChanged = (): void => {
+    if (fromAutocomplete) {
+      const geometry: any = fromAutocomplete.getPlace();
+      setFrom(geometry.formatted_address);
+      setFromTouched(() => true);
+      toTouched && createPath();
+    } else {
+      console.log('Autocomplete is not loaded yet!');
```
Comment (Pr1sm0) [inline client/src/pages/Order/Order.tsx]: Delete

File: client/src/pages/Order/Order.tsx [hunk 13]
```diff
@@ -1,13 +1,102 @@
...
+  };
+  const onToChanged = (): void => {
+    if (toAutocomplete) {
+      const geometry: any = toAutocomplete.getPlace();
+
+      setTo(geometry.formatted_address);
+      setToTouched(true);
+      fromTouched && createPath();
+    } else {
+      console.log('Autocomplete is not loaded yet!');
```
Comment (Pr1sm0) [inline client/src/pages/Order/Order.tsx]: Delete

File: client/src/pages/Order/Order.tsx [hunk 14]
```diff
@@ -1,13 +1,102 @@
...
+    to,
+    setTo,
+    onFromAutocompleteLoad,
+    onToAutocompleteLoad,
+    onFromChanged,
+    onToChanged,
+    createPath,
+    carTypes: car_types,
+    extraServices: extra_services,
+    // currentCity,
```
Comment (Pr1sm0) [inline client/src/pages/Order/Order.tsx]: Delete or uncomment if necessary

File: client/src/pages/Order/OrderForm.tsx [hunk 15]
```diff
@@ -0,0 +1,248 @@
...
+  onToAutocompleteLoad: (autocomplete: any) => void;
+  onFromChanged: () => void;
+  onToChanged: () => void;
+  setTo: (v: string) => void;
+  setFrom: (v: string) => void;
+  from: string;
+  to: string;
+  carTypes?: CarTypesI[];
+  extraServices?: ExtraServicesI[];
+  // currentCity?: string;
```
Comment (Pr1sm0) [inline client/src/pages/Order/OrderForm.tsx]: Delete or uncomment if necessary

File: client/src/pages/Order/OrderForm.tsx [hunk 16]
```diff
@@ -0,0 +1,248 @@
...
+  onFromChanged,
+  onToAutocompleteLoad,
+  onToChanged,
+  from,
+  to,
+  setFrom,
+  setTo,
+  carTypes,
+  extraServices,
+  // currentCity,
```
Comment (Pr1sm0) [inline client/src/pages/Order/OrderForm.tsx]: Delete or uncomment if necessary

File: client/src/pages/Order/OrderForm.tsx [hunk 17]
```diff
@@ -0,0 +1,248 @@
...
+              </Button>
+              <Button type="submit" variant="info">
+                Make order
+              </Button>
+            </ButtonGroup>
+          </ButtonToolbar>
+
+          {/* <button  className="btn btn-lg btn-info">
+          Calculate
+        </button> */}
```
Comment (Pr1sm0) [inline client/src/pages/Order/OrderForm.tsx]: Delete or uncomment if necessary

File: client/src/pages/Order/mapService.ts [hunk 18]
```diff
@@ -0,0 +1,91 @@
...
+        name,
+      },
+    },
+  );
+  return response;
+
+  // if (res.statusText === 'OK') {
+  //   console.log(res, 'info');
+  //   return res.data.data;
+  // }
```
Comment (Pr1sm0) [inline client/src/pages/Order/mapService.ts]: Delete or uncomment if necessary

File: client/src/reducers/mapReducer.ts [hunk 19]
```diff
@@ -0,0 +1,23 @@
+import { MapAction, MapActionTypes } from '../types/mapTypes';
+import { MapState } from './../types/mapTypes';
+
+export const initialState: MapState = {
+  map: null,
+  // renderer: null,
```
Comment (Pr1sm0) [inline client/src/reducers/mapReducer.ts]: Delete or uncomment if necessary

File: client/src/reduxStore.ts [hunk 20]
```diff
@@ -1,37 +1,55 @@
...
-import { authReducer, initialState as AuthState } from './reducers/authReducer';
 import createSagaMiddleware from 'redux-saga';
-import { rootWatcher } from './sagas/index';
+import { composeWithDevTools } from 'redux-devtools-extension';
 import { createBrowserHistory } from 'history';
 import { connectRouter } from 'connected-react-router';
 import { routerMiddleware } from 'react-router-redux';
 
-export const history = createBrowserHistory();
+// import { userReducer, initialState as UserState } from './reducers/userReducer';
```
Comment (Pr1sm0) [inline client/src/reduxStore.ts]: Delete or uncomment if necessary

File: client/src/reduxStore.ts [hunk 21]
```diff
@@ -1,37 +1,55 @@
...
+} from './reducers/orderReducer';
+import { authReducer, initialState as AuthState } from './reducers/authReducer';
+import { rootWatcher } from './sagas/index';
 
-const browserMiddleware = routerMiddleware(history);
+const sagaMiddleware = createSagaMiddleware();
 
 const initialState = {
   auth: AuthState,
+  // user: UserState,
```
Comment (Pr1sm0) [inline client/src/reduxStore.ts]: Delete or uncomment if necessary

File: client/src/reduxStore.ts [hunk 22]
```diff
@@ -1,37 +1,55 @@
...
 
-const sagaMiddleware = createSagaMiddleware();
+export const history = createBrowserHistory();
+const browserMiddleware = routerMiddleware(history);
 
 export const rootReducer = (history: any) =>
   combineReducers({
     router: connectRouter(history),
     auth: authReducer,
+    // user: userReducer,
```
Comment (Pr1sm0) [inline client/src/reduxStore.ts]: Delete or uncomment if necessary

File: client/src/sagas/infoSaga.ts [hunk 23]
```diff
@@ -0,0 +1,24 @@
...
+
+function* fetchCityInfoWorker(): Generator<StrictEffect, void, any> {
+  const cityInfoState = yield select(getCityNameFromState);
+  console.log('cityInfoState', cityInfoState);
+
+  const data = (yield call(
+    fetchCityInfo(cityInfoState.name),
+  )) as AxiosResponse<CityInfoI>;
+
+  console.log(data, 'data');
```
Comment (Pr1sm0) [inline client/src/sagas/infoSaga.ts]: Delete

File: client/src/sagas/infoSaga.ts [hunk 24]
```diff
@@ -0,0 +1,24 @@
...
+import { InfoActionTypes } from '../types/cityInfoTypes';
+import { fetchCityInfo, CityInfoI } from './../pages/Order/mapService';
+import { setCityInfoCreator } from '../actions/cityInfoActions';
+import { AxiosResponse } from 'axios';
+
+export const getCityNameFromState = (state: any) => state.cityInfo;
+
+function* fetchCityInfoWorker(): Generator<StrictEffect, void, any> {
+  const cityInfoState = yield select(getCityNameFromState);
+  console.log('cityInfoState', cityInfoState);
```
Comment (Pr1sm0) [inline client/src/sagas/infoSaga.ts]: Delete

File: client/src/sagas/mapSaga.ts [hunk 25]
```diff
@@ -0,0 +1,59 @@
...
+  };
+};
+
+const googleApiKey = process.env.REACT_APP_MAP_API_KEY;
+
+export const getCityNameFromState = (state: any) => state.cityInfo;
+
+const getCityName = (l: any) => () => {
+  console.log('key api', process.env.REACT_APP_MAP_API_KEY!);
+  console.log('key api', googleApiKey);
```
Comment (Pr1sm0) [inline client/src/sagas/mapSaga.ts]: Delete

File: client/src/sagas/mapSaga.ts [hunk 26]
```diff
@@ -0,0 +1,59 @@
...
+  const name = res.data.plus_code.compound_code.split(', ')[1].split(' ')[0];
+  yield put(changeCityInfoValueCreator('name', name));
+  yield put(getCityInfoCreator(name));
+  console.log('cityName', name);
+}
+
+// function* getCityNameFromLocation(): Generator {
+//   const res = yield call(getCityName);
+
+// }
```
Comment (Pr1sm0) [inline client/src/sagas/mapSaga.ts]: Delete or uncomment if necessary

File: client/src/types/mapTypes.ts [hunk 27]
```diff
@@ -0,0 +1,36 @@
+export interface CurrentLocation {
+  lat: number;
+  lng: number;
+}
+
+export interface MapState {
+  map: google.maps.Map | null;
+  // renderer: any;
+  // renderer: google.maps.DirectionsRenderer;
```
Comment (Pr1sm0) [inline client/src/types/mapTypes.ts:9]: Delete or uncomment if necessary

File: client/tsconfig.json [hunk 28]
```diff
@@ -15,6 +15,8 @@
     "isolatedModules": true,
     "noEmit": true,
     "jsx": "react-jsx"
+    // "types": ["google.maps"]
```
Comment (Pr1sm0) [inline client/tsconfig.json:18]: Delete or uncomment if necessary

File: client/tsconfig.json [hunk 29]
```diff
@@ -15,6 +15,8 @@
...
   },
   "include": ["src"]
+  // "files": ["./index.d.ts"]
```
Comment (Pr1sm0) [inline client/tsconfig.json:21]: Delete or uncomment if necessary

File: server/src/app.ts [hunk 30]
```diff
@@ -47,17 +54,34 @@ if (process.env.NODE_ENV !== PRODUCTION) {
...
+    const options = {
+      force: true,
+      // alter: true,
+    };
+    await sequelize.sync(options);
+    // Possibly not right
+
+    // Uncomment if you don't have city table
+    // and set optins to alter: true
+    // comment after creating in db and disable alter (comment it)
```
Comment (Pr1sm0) [inline server/src/app.ts]: If there are no tables, we can use sync with no parameters to create tables

File: server/src/app.ts [hunk 31]
```diff
@@ -47,17 +54,34 @@ if (process.env.NODE_ENV !== PRODUCTION) {
...
+        basePrice: 41,
+        basePriceForKm: 10,
+        car_types: carTypes.slice(0, 4),
+        extra_services: extraServices,
+      },
+      {
+        include: [sequelize.models[CAR_TYPE], sequelize.models[EXTRA_SERVICE]],
+      },
+    );
+    // End Possibly not right
```
Comment (Pr1sm0) [inline server/src/app.ts:86]: Delete

File: server/src/controllers/authController.ts [hunk 32]
```diff
@@ -1,7 +1,7 @@
-import sequelize from '../db/sequelize/models/index';
-import * as jwt from 'jsonwebtoken';
-import * as bcrypt from 'bcrypt';
 import { NextFunction, Request, Response } from 'express';
+import * as bcrypt from 'bcrypt';
+import * as jwt from 'jsonwebtoken';
+import sequelize from '../db/sequelize/models/index';
 import ApiError from '../errors/ApiErrors';
 // import { MobizonResponse, sendSMS } from '../services/notification';
```
Comment (Pr1sm0) [inline server/src/controllers/authController.ts]: Delete or uncomment if necessary

File: server/src/controllers/authController.ts [hunk 33]
```diff
@@ -15,15 +15,14 @@ import { generateVerifyCode } from '../services/verification';
 // TODO CONSTANTS FOR MODELS
```
Comment (Pr1sm0) [inline server/src/controllers/authController.ts]: Delete

File: server/src/controllers/authController.ts [hunk 34]
```diff
@@ -54,11 +53,15 @@ export default class authController {
...
+              user.name,
+              authorities,
+            );
             const refreshToken = generateRefreshToken(
               user.id,
               user.name,
-              authorities
+              authorities,
             );
             // const noteServiceRes: Promise<MobizonResponse> = sendSMS(
```
Comment (Pr1sm0) [inline server/src/controllers/authController.ts]: Delete or uncomment if necessary

File: server/src/controllers/authController.ts [hunk 35]
```diff
@@ -170,33 +181,30 @@ export default class authController {
...
-  async delToken (
-    req: Request,
-    res: Response,
-  ): Promise<any> {
-    const {refreshToken} = req.cookies;
+  async delToken(req: Request, res: Response): Promise<any> {
+    const { refreshToken } = req.cookies;
     deleteToken(req.body, refreshToken);
     res.clearCookie('refreshToken');
     res.sendStatus(204);
```
Comment (Pr1sm0) [inline server/src/controllers/authController.ts:253]: Status to contant

File: server/src/controllers/infoController.ts [hunk 36]
```diff
@@ -0,0 +1,32 @@
+import { Request, Response } from 'express';
+import sequelize from '../db/sequelize/models/index';
+import { STATUS_BAD_REQUEST, STATUS_OK } from '../constants/api';
+import { CAR_TYPE, CITY, EXTRA_SERVICE } from '../constants/modelsNames';
+
+export default class InfoController {
+  get = async (req: Request, res: Response): Promise<any> => {
+    const { name } = req.query;
+    console.log(name, 'name');
```
Comment (Pr1sm0) [inline server/src/controllers/infoController.ts]: Delete

File: server/src/controllers/orderController.ts [hunk 37]
```diff
@@ -0,0 +1,48 @@
...
+        message: error,
+        status: STATUS_BAD_REQUEST,
+      });
+    }
+  };
+
+  get = async (req: Request, res: Response): Promise<any> => {
+    let { limit, page } = req.query;
+    const { where } = req.query;
+    // res.json(where);
```
Comment (Pr1sm0) [inline server/src/controllers/orderController.ts:23]: Delete or uncomment if necessary

File: server/src/controllers/orderController.ts [hunk 38]
```diff
@@ -0,0 +1,48 @@
...
+
+    try {
+      const data = await sequelize.models[ORDER].findAndCountAll({
+        where,
+        offset,
+        limit,
+      });
+      res.status(STATUS_OK).send({ data, status: STATUS_OK });
+    } catch (error) {
+      console.log(error);
```
Comment (Pr1sm0) [inline server/src/controllers/orderController.ts:39]: Delete

File: server/src/db/sequelize/migrations/20210605154319-create-order.js [hunk 39]
```diff
@@ -0,0 +1,67 @@
...
+        primaryKey: true,
+        type: Sequelize.INTEGER,
+      },
+      customer_id: {
+        type: Sequelize.INTEGER,
+        allowNull: false,
+        // references: {
+        //   model: 'user',
+        //   key: 'id',
+        // },
```
Comment (Pr1sm0) [inline server/src/db/sequelize/migrations/20210605154319-create-order.js:16]: Delete or uncomment if necessary

File: server/src/db/sequelize/migrations/20210605154319-create-order.js [hunk 40]
```diff
@@ -0,0 +1,67 @@
...
+      },
+      driver_id: {
+        type: Sequelize.INTEGER,
+        allowNull: true,
+        // references: {
+        //   model: 'driver',
+        //   key: 'id',
+        // },
```
Comment (Pr1sm0) [inline server/src/db/sequelize/migrations/20210605154319-create-order.js:24]: Delete or uncomment if necessary

File: server/src/db/sequelize/models/carType.ts [hunk 41]
```diff
@@ -0,0 +1,45 @@
...
+  CITY_CAR_TYPES,
+} from '../../../constants/modelsNames';
+
+export default (sequelize: any, DataTypes: any) => {
+  class CarType extends Model {
+    /**
+     * Helper method for defining associations.
+     * This method is not a part of Sequelize lifecycle.
+     * The `models/index` file will call this method automatically.
+     */
```
Comment (Pr1sm0) [inline server/src/db/sequelize/models/carType.ts]: Delete

File: server/src/db/sequelize/models/carType.ts [hunk 42]
```diff
@@ -0,0 +1,45 @@
...
+    static associate(models: any) {
+      // define association here
+      // CarType.hasMany(models[CITY]);
```
Comment (Pr1sm0) [inline server/src/db/sequelize/models/carType.ts]: Delete or uncomment if necessary

File: server/src/db/sequelize/models/city.ts [hunk 43]
```diff
@@ -0,0 +1,76 @@
...
+} from '../../../constants/modelsNames';
+import { extraServices } from '../../../constants/seeders';
+
+export default (sequelize: any, DataTypes: any) => {
+  class City extends Model {
+    /**
+     * Helper method for defining associations.
+     * This method is not a part of Sequelize lifecycle.
+     * The `models/index` file will call this method automatically.
+     */
```
Comment (Pr1sm0) [inline server/src/db/sequelize/models/city.ts]: Delete

File: server/src/db/sequelize/models/city.ts [hunk 44]
```diff
@@ -0,0 +1,76 @@
...
+    static associate(models: any) {
+      // City.hasMany(models[CAR_TYPE]);
```
Comment (Pr1sm0) [inline server/src/db/sequelize/models/city.ts]: Delete or uncomment if necessary

File: server/src/db/sequelize/models/city.ts [hunk 45]
```diff
@@ -0,0 +1,76 @@
...
+      //     name: 'Чернівці',
+      //     basePrice: 40,
+      //     basePriceForKm: 10,
+      //     car_types: carTypes.slice(0, 4),
+      //   },
+      //   {
+      //     include: [sequelize.models.car_type],
+      //   },
+      // );
+      // console.log('City created');
```
Comment (Pr1sm0) [inline server/src/db/sequelize/models/city.ts]: Delete or uncomment if necessary

File: server/src/db/sequelize/models/city.ts [hunk 46]
```diff
@@ -0,0 +1,76 @@
...
+      basePriceForKm: {
+        type: DataTypes.INTEGER,
+        allowNull: true,
+      },
+    },
+    {
+      sequelize,
+      modelName: CITY,
+      underscored: true,
+      // hooks: {},
```
Comment (Pr1sm0) [inline server/src/db/sequelize/models/city.ts]: Delete or uncomment if necessary

File: server/src/db/sequelize/models/citycartype.ts [hunk 47]
```diff
@@ -0,0 +1,30 @@
+import { Model } from 'sequelize';
+import { CITY_CAR_TYPES } from '../../../constants/modelsNames';
+
+export default (sequelize: any, DataTypes: any) => {
+  class cityCarType extends Model {
+    /**
+     * Helper method for defining associations.
+     * This method is not a part of Sequelize lifecycle.
+     * The `models/index` file will call this method automatically.
+     */
```
Comment (Pr1sm0) [inline server/src/db/sequelize/models/citycartype.ts]: Delete

File: server/src/db/sequelize/models/citycartype.ts [hunk 48]
```diff
@@ -0,0 +1,30 @@
...
+    static associate(models: any) {
+      // define association here
```
Comment (Pr1sm0) [inline server/src/db/sequelize/models/citycartype.ts]: Delete

File: server/src/db/sequelize/models/cityservices.ts [hunk 49]
```diff
@@ -0,0 +1,30 @@
+import { Model } from 'sequelize';
+import { CITY_SERVICES } from '../../../constants/modelsNames';
+
+export default (sequelize: any, DataTypes: any) => {
+  class cityServices extends Model {
+    /**
+     * Helper method for defining associations.
+     * This method is not a part of Sequelize lifecycle.
+     * The `models/index` file will call this method automatically.
+     */
```
Comment (Pr1sm0) [inline server/src/db/sequelize/models/cityservices.ts]: Delete

File: server/src/db/sequelize/models/cityservices.ts [hunk 50]
```diff
@@ -0,0 +1,30 @@
...
+    static associate(models: any) {
+      // define association here
```
Comment (Pr1sm0) [inline server/src/db/sequelize/models/cityservices.ts]: Delete

File: server/src/db/sequelize/models/creditCard.ts [hunk 51]
```diff
@@ -0,0 +1,43 @@
+import { Model } from 'sequelize';
+import { CREDIT_CARD } from '../../../constants/modelsNames';
+
+export default (sequelize: any, DataTypes: any) => {
+  class CreditCard extends Model {
+    /**
+     * Helper method for defining associations.
+     * This method is not a part of Sequelize lifecycle.
+     * The `models/index` file will call this method automatically.
+     */
```
Comment (Pr1sm0) [inline server/src/db/sequelize/models/creditCard.ts]: Delete

File: server/src/db/sequelize/models/creditCard.ts [hunk 52]
```diff
@@ -0,0 +1,43 @@
...
+    static associate() {
+      // define association here
```
Comment (Pr1sm0) [inline server/src/db/sequelize/models/creditCard.ts]: Delete

File: server/src/db/sequelize/models/driver.ts [hunk 53]
```diff
@@ -7,17 +8,22 @@ export default (sequelize: any, DataTypes: any) => {
      * This method is not a part of Sequelize lifecycle.
      * The `models/index` file will call this method automatically.
      */
```
Comment (Pr1sm0) [inline server/src/db/sequelize/models/driver.ts]: Delete

File: server/src/db/sequelize/models/driver.ts [hunk 54]
```diff
@@ -7,17 +8,22 @@ export default (sequelize: any, DataTypes: any) => {
...
-    static associate() {
-      // define association here 
+    static associate(models: any) {
+      // define association here
       // sequelize.models['users'].belongsTo(Driver)
```
Comment (Pr1sm0) [inline server/src/db/sequelize/models/driver.ts]: Delete or uncomment if necessary

File: server/src/db/sequelize/models/extraService.ts [hunk 55]
```diff
@@ -0,0 +1,42 @@
...
+  EXTRA_SERVICE,
+} from '../../../constants/modelsNames';
+
+export default (sequelize: any, DataTypes: any) => {
+  class ExtraService extends Model {
+    /**
+     * Helper method for defining associations.
+     * This method is not a part of Sequelize lifecycle.
+     * The `models/index` file will call this method automatically.
+     */
```
Comment (Pr1sm0) [inline server/src/db/sequelize/models/extraService.ts]: Delete

File: server/src/db/sequelize/models/extraService.ts [hunk 56]
```diff
@@ -0,0 +1,42 @@
...
+    static associate(models: any) {
+      // define association here
```
Comment (Pr1sm0) [inline server/src/db/sequelize/models/extraService.ts]: Delete

File: server/src/db/sequelize/models/order.ts [hunk 57]
```diff
@@ -0,0 +1,81 @@
...
+import { CAR_TYPE, DRIVER, ORDER, USER } from '../../../constants/modelsNames';
+import { USER_IN_ORDER } from '../../../constants/foreignKeys';
+
+export default (sequelize: any, DataTypes: any) => {
+  class Order extends Model {
+    /**
+     * Helper method for defining associations.
+     * This method is not a part of Sequelize lifecycle.
+     * The `models/index` file will call this method automatically.
+     */
```
Comment (Pr1sm0) [inline server/src/db/sequelize/models/order.ts]: Delete

File: server/src/db/sequelize/models/order.ts [hunk 58]
```diff
@@ -0,0 +1,81 @@
...
+        },
+      });
+      Order.belongsTo(models[DRIVER]);
+      Order.belongsTo(models[CAR_TYPE]);
+      // define association here
+      // create associate with user and driver(id for user and id for driver from driver)
+      // order.hasOne(carType)
+      // carType.belongsTo(Order)
+      // order.hasMany(extraServices)
+      // extraServices.belongsToMany(Order)
```
Comment (Pr1sm0) [inline server/src/db/sequelize/models/order.ts]: Delete or uncomment if necessary

File: server/src/db/sequelize/models/order.ts [hunk 59]
```diff
@@ -0,0 +1,81 @@
...
+        autoIncrement: true,
+      },
+      // customer_id: {
+      //   type: DataTypes.INTEGER,
+      //   allowNull: false,
+      // },
+      // driver_id: {
+      //   type: DataTypes.INTEGER,
+      //   allowNull: true,
+      // },
```
Comment (Pr1sm0) [inline server/src/db/sequelize/models/order.ts]: Delete or uncomment if necessary

File: server/src/db/sequelize/models/order.ts [hunk 60]
```diff
@@ -0,0 +1,81 @@
...
+        allowNull: false,
+      },
+      price: {
+        type: DataTypes.STRING,
+        allowNull: false,
+      },
+      // car_type: {
+      //   type: DataTypes.STRING,
+      //   allowNull: false,
+      // },
```
Comment (Pr1sm0) [inline server/src/db/sequelize/models/order.ts]: Delete or uncomment if necessary

File: server/src/db/sequelize/models/token.ts [hunk 61]
```diff
@@ -7,9 +7,7 @@ export default (sequelize: any, DataTypes: any) => {
      * This method is not a part of Sequelize lifecycle.
      * The `models/index` file will call this method automatically.
      */
```
Comment (Pr1sm0) [inline server/src/db/sequelize/models/token.ts]: Delete

File: server/src/db/sequelize/models/user.ts [hunk 62]
```diff
@@ -7,8 +9,12 @@ export default (sequelize: any, DataTypes: any) => {
      * This method is not a part of Sequelize lifecycle.
      * The `models/index` file will call this method automatically.
      */
```
Comment (Pr1sm0) [inline server/src/db/sequelize/models/user.ts]: Delete

File: server/src/db/sequelize/seeders/20210605162802-CarType.js [hunk 63]
```diff
@@ -0,0 +1,43 @@
...
+});
+
+module.exports = {
+  up: async (queryInterface, Sequelize) => {
+    /**
+     * Add seed commands here.
+     *
+     * Example:
+     *
+     */
```
Comment (Pr1sm0) [inline server/src/db/sequelize/seeders/20210605162802-CarType.js:31]: Delete

File: server/src/db/sequelize/seeders/20210605162802-CarType.js [hunk 64]
```diff
@@ -0,0 +1,43 @@
...
+    await queryInterface.bulkInsert('car_types', carTypes, {});
+  },
+
+  down: async (queryInterface, Sequelize) => {
+    /**
+     * Add commands to revert seed here.
+     *
+     * Example:
+     * await queryInterface.bulkDelete('People', null, {});
+     */
```
Comment (Pr1sm0) [inline server/src/db/sequelize/seeders/20210605162802-CarType.js:41]: Delete

File: server/src/db/sequelize/seeders/20210605164610-ExtraServices.js [hunk 65]
```diff
@@ -0,0 +1,43 @@
...
+  up: async (queryInterface, Sequelize) => {
+    /**
+     * Add seed commands here.
+     *
+     * Example:
+     * await queryInterface.bulkInsert('People', [{
+     *   name: 'John Doe',
+     *   isBetaMember: false
+     * }], {});
+    */
```
Comment (Pr1sm0) [inline server/src/db/sequelize/seeders/20210605164610-ExtraServices.js:30]: Delete

File: server/src/db/sequelize/seeders/20210605164610-ExtraServices.js [hunk 66]
```diff
@@ -0,0 +1,43 @@
...
+    await queryInterface.bulkInsert('extra_services', extraServices, {});
+  },
+
+  down: async (queryInterface, Sequelize) => {
+    /**
+     * Add commands to revert seed here.
+     *
+     * Example:
+     * await queryInterface.bulkDelete('People', null, {});
+     */
```
Comment (Pr1sm0) [inline server/src/db/sequelize/seeders/20210605164610-ExtraServices.js:41]: Delete

File: server/src/db/sequelize/seeders/20210605170547-City.js [hunk 67]
```diff
@@ -0,0 +1,56 @@
...
+  up: async (queryInterface, Sequelize) => {
+    /**
+     * Add seed commands here.
+     *
+     * Example:
+     * await queryInterface.bulkInsert('People', [{
+     *   name: 'John Doe',
+     *   isBetaMember: false
+     * }], {});
+    */
```
Comment (Pr1sm0) [inline server/src/db/sequelize/seeders/20210605170547-City.js:41]: Delete

File: server/src/db/sequelize/seeders/20210605170547-City.js [hunk 68]
```diff
@@ -0,0 +1,56 @@
...
+    });
+  },
+
+  down: async (queryInterface, Sequelize) => {
+    /**
+     * Add commands to revert seed here.
+     *
+     * Example:
+     * await queryInterface.bulkDelete('People', null, {});
+     */
```
Comment (Pr1sm0) [inline server/src/db/sequelize/seeders/20210605170547-City.js:53]: Delete

File: server/src/utils/jwtHelpers.ts [hunk 69]
```diff
@@ -1,46 +1,54 @@
 import * as jwt from 'jsonwebtoken';
 import sequelize from '../db/sequelize/models/index';
-import ApiError from '../errors/ApiErrors';
+// import ApiError from '../errors/ApiErrors';
```
Comment (Pr1sm0) [inline server/src/utils/jwtHelpers.ts]: Delete or uncomment if necessary

File: server/src/utils/jwtHelpers.ts [hunk 70]
```diff
@@ -1,46 +1,54 @@
...
-       process.env.REFRESH_TOKEN_SECRET_KEY,
-       {expiresIn: process.env.REFRESH_TOKEN_TIME})
-       refreshTokens.push(refreshToken)
-       return refreshToken
+export const generateAccessToken = (
+  id: number,
+  name: string,
+  roles: string[],
+) => {
+  console.log({ id, name, roles });
```
Comment (Pr1sm0) [inline server/src/utils/jwtHelpers.ts]: Delete

File: server/src/utils/jwtHelpers.ts [hunk 71]
```diff
@@ -1,46 +1,54 @@
...
+    { expiresIn: process.env.REFRESH_TOKEN_TIME },
+  );
+  refreshTokens.push(refreshToken);
+  return refreshToken;
+};
+
+export const deleteToken = (body: any, refreshToken: string) => {
+  refreshTokens = refreshTokens.filter((token) => token !== body.token);
+  // const tokenData = Token.deleteOne({refreshToken})
+  // return tokenData;
```
Comment (Pr1sm0) [inline server/src/utils/jwtHelpers.ts]: Delete or uncomment if necessary

File: server/src/utils/jwtHelpers.ts [hunk 72]
```diff
@@ -1,46 +1,54 @@
...
-        return tokenData.save();
-    }
-    const token = Token.create({user_id: userId, refreshToken})
-        return token
-}
\ No newline at end of file
+  const tokenData = Token.findOne({ where: { user_id: userId } });
+  if (tokenData) {
+    tokenData.refreshToken = refreshToken;
+    console.log('TOKEN DATA', tokenData);
```
Comment (Pr1sm0) [inline server/src/utils/jwtHelpers.ts:47]: Delete

**Review by Pr1sm0 — APPROVED** [review]

**Review by N-Balinskiy — APPROVED** [review]

**Pull request closed without merging**